Crosstabs

gregmacfarlane opened the pull request:
This adds functionality for publication-ready cross-tabulations. The function determines, for instance, if a variable is numeric or character and labels the rows and columns appropriately. The function also allows the user to specify if they want marginal totals appended to the table, or if the table should be presented as percents. Weighting the table by a third variable in the table is supported.

gregmacfarlane committed:
Fix bug in mdd x-axis

The single and colored plots need to use the same x-axis.

gregmacfarlane committed:
Update links with mixed area type example

gregmacfarlane committed:
Use factors instead of characters

This should close #10; it requires that text fields either have explicity factor levels or it will return alphabetical order. But not that bad.

gregmacfarlane committed:
Add descriptive area type name to links

gregmacfarlane committed:
Import tidyr

gregmacfarlane committed:
Add crosstable function

This function is a generic cross-tabulation function like 
`table`, but it contains options to weight the data and 
to place marginals on the sides. It also labels the 
columns descriptively and gives an option to present
the table as percentages. And because the output is a data_frame, it can be wrapped with kable() for print-quality output.

This was migrated from the (somewhat stale) nhtsHelper repository.

gregmacfarlane committed:
Use standard evaluation

This helps avoid NOTEs in R CMD CHECK

gregmacfarlane committed:
Fix prepending behavior

Previously the variable name would be prepended to both the row and column names. This is good when the variable field is ambiguous (1, 2, 3, ...) but bad when the field is already a factor or a character variable. This commit allows the function to determine what is happening independently.

gregmacfarlane committed:
Add crosstab to template

gregmacfarlane commented:
This pull request addresses #12

gregmacfarlane committed:
Restructure link assignment tables

Can reduce code duplication with a single table function that can handle both flow and RMSE statistics. The user selects which to use with a function argument.

Additionally, restructured gave a volume_breaks argument to allow the volume groups to be specified by the user.

gregmacfarlane committed:
Update links table

Add the factored and named area type group to the table.

gregmacfarlane committed:
Restructure template for the new function call

gregmacfarlane committed:
Repair node ids

gregmacfarlane committed:
Merge pull request #15 from pbsag/assignment_tables

Assignment tables

gregmacfarlane committed:
Use count as x axis on mdd plot

Commit:
Merge pull request #18 from pbsag/hotfix/mdd

Use count as x axis on mdd plot

gregmacfarlane committed:
Export helper functions

I use these in many other places, so we'll keep them available.

gregmacfarlane committed:
Add function to numerify a mixed vector

gregmacfarlane committed:
Increment version

Commit:
Merge pull request #20 from pbsag/feature/helpers

Helpers

gregmacfarlane committed:
Make a plotly mdd plot

gregmacfarlane committed:
Make a plotly validation plot

Commit:
Update links with distance field

Commit:
Changes suggested by roxygen2

Commit:
Update raw data and Links with fields "speed",v "ffspeed", "capacity"

Commit:
Add functions to calculate VMT, VHT, VHD and VOC

Commit:
Rebuild the documentation with added functions and data frame

Commit:
Simplify code

Only need to include the aggregation function in the ifelse tree; everything else can go at the end. This will make fixing and improving simpler later on.

Commit:
Make fields optional

If I'm calculating VMT, I don't need to supply a capacity field.

Commit:
Fix namespace

gregmacfarlane committed:
Merge pull request #21 from pbsag/feature/vmt

Feature/vmt

gregmacfarlane committed:
Increment version

gregmacfarlane committed:
Merge pull request #23 from pbsag/gregmacfarlane-patch-1

Increment version

Commit:
RE-code voc with weighted average of volume

Commit:
Rebuild rox with development version

gregmacfarlane committed:
Merge pull request #25 from pbsag/hotfix/voc

RE-code voc with weighted average of volume

Commit:
Increment version and document package

gregmacfarlane committed:
Merge pull request #26 from pbsag/version/0.0.5

Increment version and document package

Commit:
Correct plot_mdd

Remove the stat smoothing which doesn't really make any
sense in this context. Also make the mdd range a ribbon behind the objects, and show positive/negative error instead of just absolute error.

Commit:
Only convert to factor if not already

If the user gives a numeric or integer value as `color_field`, ggplot2 will only work properly if we convert that field to a color. #22 pointed out that if a variable is already a factor, it might be best to leave it alone.

This introduces a pass such that variables are only factored if they are not already factors.

Commit:
Add package dependencies

Commit:
Update travis instructions

gregmacfarlane committed:
Merge branch 'master' into feature/plotly

gregmacfarlane committed:
Merge pull request #28 from pbsag/feature/plotly

Feature/plotly

Commit:
Add flows dataset for testing and example building

Commit:
Add function to show chord chart

This new function is simply a wrapper to https://github.com/mattflor/chorddiag so that the user can pass in a table object instead of a matrix.

Commit:
Update with "count" rather than "volume" as the "group_field" when doing RMSE or Count/Flow comparisons

gregmacfarlane committed:
Merge pull request #30 from pbsag/count/volume

Update with "count" rather than "volume" as the "group_field" when do…

gregmacfarlane committed:
Add local chorddiag to function documentation

Commit:
Add tidyr to package namespace

gregmacfarlane committed:
Merged branch master into feature/chords

gregmacfarlane committed:
Merge pull request #29 from pbsag/feature/chords

Feature/chords

Commit:
Replace R2 with PRMSE

R2 is actually not an appropriate measurement for this
case. We should use overall PRMSE instead. Close #19

gregmacfarlane committed:
Remove smooth from validation plots

It really doesn't add to interpretation. Addresses #22

Commit:
Add link ID to plotly graphs

The user can specify an ID column, else it will use the row names already on the data frame.

Commit:
Rebuild rox

gregmacfarlane committed:
Merge pull request #34 from pbsag/hotfix/labels

Hotfix/labels

Commit:
Remove the mdd.rda file

Also remove the entry for it in data.R

Commit:
Add the mdd_table() function

Given from, to, and step arguments, generates
a table to of volumes and their respective
MDD.

Commit:
Update mdd_table()

Correct the case_when() equations to give the right results.
Update the default input values.

Commit:
Add mdd_table() to plot_mdd() and plotly_mdd()

Commit:
Update roxygen documentation

Commit:
Add description to mdd_table()

gregmacfarlane committed:
Merge pull request #37 from pbsag/mdd_table

Replaces static lookup table with exposed dynamic function to calculate MDD curves.

Commit:
Improve plotly_validation()

Now supports a regression line and equation.
Also shows link IDs appropriately and handles
a null "id" variable correctly.

Commit:
Merge pull request #38 from pbsag/feature/plotly

Improve plotly_validation()

Commit:
Merge branch 'crosstabs'

# Conflicts:
#	DESCRIPTION
#	NAMESPACE
#	R/data.R
#	data-raw/links.R
#	data/links.rda
#	inst/rmarkdown/templates/assignment_validation/skeleton/skeleton.Rmd
#	man/links.Rd

dkyleward commented:
Had to do some manual conflict resolution and regenerating the package documentation using roxygen. I merged locally, but couldn't push without the travis checks. Instead, I merged the resolutions back into this branch to update the pull request.

Commit:
Add distance field to links table

Lost this field during conflict resolution, which led to
errors with the link_measures_table() function.
Added a distance field was placeholder value.

Commit:
Add speed and ffspeed fields to links table

Commit:
Correct links table

Took links.rda from an earlier commit on the master branch.
The crosstab function required at least one link have an area
type of "CBD", so I made that change. This is because the field
is defined as a factor with five levels, one of which is "CBD".
